litgpt/tokenizer.py

```diff
@@ -97,7 +101,8 @@ def encode(
             bos_id = self.bos_id
             if bos_id is None:
                 raise NotImplementedError("This tokenizer does not have a defined a bos token")
-            tokens = [bos_id] + tokens
+            if tokens[0] != bos_id:
+                tokens = [bos_id] + tokens
         if eos:
             tokens = tokens + [self.eos_id]
```

Comment:
Should we add a check similar to the one above?

```suggestion
        if tokens and tokens[-1] != eos_id:
            tokens = tokens + [eos_id]
```

Reply:
The issue was only with `bos` token.
But I guess it won't hurt.

Could you do the honors?
I'm already away from my laptop.

Reply:
Just change your code a bit.
You want to add eos token only if it's set to True and it hasn't been added already.

Reply:
Sounds good. What about the following update code (above) then?

Reply:
`not tokens` is True for an empty list but also for a None value, so that means that the code might try to sum None and a list (if tokens is None), which will result in an error.

Reply:
Ah my bad, I thought it would be an empty list. But then we would have that issue in the original code,

```
if eos:
    tokens = tokens + [self.eos_id]
```

too?

Reply:
True. Keep your code, but add also a check that the processor didn't return a None value. If it's so, then it's definitely an error.

Reply:
You mean adding an

```
if tokens is None:
    raise ValueError("`tokens` is None")
```

?

Reply:
Yes 👍